### PR TITLE
Integrate Algolia search

### PR DESCRIPTION
**Pull request by settermjd**
After much research, using a combination of Algolia and their docsearch-scraper Docker image to scrape the documentation and maintain the search index seems to be the best approach. Given that, this PR:

- Removes all references to lunr and antora-lunr (which was how things worked before)
- Updates the documentation to reflect the new strategy
- Adds a new Drone pipeline item

@tboerger, I need help to finish the Drone pipeline item please, as I'm sure it's not complete, and I need to get the secrets added to the configuration, which I can't do on my own.

### CODE REVIEW

**Comment by mmattel** [comment]
It would be great if there would be a make command to update the search index like we have make html.

**Comment by settermjd** [comment]
@tboerger, I've made some further changes to the pipeline item in .drone.yml and added in the relevant secrets. Just waiting to see if the build succeeds this time (_or at least does better than before_).

**Comment by mmattel** [comment]
Docker tells on update the search index task:
```
Virtualenv location: /root/.local/share/virtualenvs/src-x3FbBUl6
Creating a Pipfile for this project...
/root/.local/share/virtualenvs/src-x3FbBUl6/bin/python: No module named src
```

File: .drone.yml [hunk 1]
```diff
@@ -84,6 +84,25 @@ pipeline:
...
+  update-search-index:
+    image: algolia/docsearch-scraper
+    entrypoint: ""
+    pull: true
+    secrets: [ 
+      algolia_app_id, algolia_config, 
+      algolia_index_name, algolia_search_api_key, 
+      algolia_write_api_key 
+    ]
+    environment:
```
Comment (tboerger) [inline .drone.yml]: Drop most of the environment block, secrets are injected as environment variables.

File: .drone.yml [hunk 2]
```diff
@@ -84,6 +84,25 @@ pipeline:
...
+      - ALGOLIA_APP_ID=${ALGOLIA_APP_ID}
+      - ALGOLIA_API_KEY=${ALGOLIA_SEARCH_API_KEY}
+      - APPLICATION_ID=${ALGOLIA_APP_ID}
+      - API_KEY=${ALGOLIA_WRITE_API_KEY}
+      - ALGOLIA_INDEX_NAME=${ALGOLIA_INDEX_NAME}
+      - CONFIG=${ALGOLIA_CONFIG}
```
Comment (tboerger) [inline .drone.yml]: Keep this var, but directly assign the value, e.g. `CONFIG=algolia-config.json`. BTW, we should move the config to `search/algolia.json` :)

File: .drone.yml [hunk 3]
```diff
@@ -84,6 +84,25 @@ pipeline:
       local: false
       event: [ push ]
 
+  update-search-index:
+    image: algolia/docsearch-scraper
+    entrypoint: ""
```
Comment (tboerger) [inline .drone.yml]: Drop that, Drone anyway overwrites the entrypoint.

File: .drone.yml [hunk 4]
```diff
@@ -84,6 +84,25 @@ pipeline:
       local: false
       event: [ push ]
 
+  update-search-index:
+    image: algolia/docsearch-scraper
```
Comment (tboerger) [inline .drone.yml]: Always use specific tags of used images, even if there is only `latest` available be more specific.

File: .drone.yml [hunk 5]
```diff
@@ -84,6 +84,25 @@ pipeline:
...
+    environment:
+      - ALGOLIA_APP_ID=${ALGOLIA_APP_ID}
+      - ALGOLIA_API_KEY=${ALGOLIA_SEARCH_API_KEY}
+      - APPLICATION_ID=${ALGOLIA_APP_ID}
+      - API_KEY=${ALGOLIA_WRITE_API_KEY}
+      - ALGOLIA_INDEX_NAME=${ALGOLIA_INDEX_NAME}
+      - CONFIG=${ALGOLIA_CONFIG}
+    commands:
+      - pipenv run python -m src.index
+
```
Comment (tboerger) [inline .drone.yml]: You should also add a `when` block, this should only be done on `push` event.

File: .drone.yml [hunk 6]
```diff
@@ -92,3 +111,5 @@ pipeline:
       local: false
       event: [ push ]
       status: [ success, failure ]
+
```
Comment (tboerger) [inline .drone.yml]: Drop the additional blank lines?

File: .drone.yml [hunk 7]
```diff
@@ -84,6 +84,25 @@ pipeline:
...
+    ]
+    environment:
+      - ALGOLIA_APP_ID=${ALGOLIA_APP_ID}
+      - ALGOLIA_API_KEY=${ALGOLIA_SEARCH_API_KEY}
+      - APPLICATION_ID=${ALGOLIA_APP_ID}
+      - API_KEY=${ALGOLIA_WRITE_API_KEY}
+      - ALGOLIA_INDEX_NAME=${ALGOLIA_INDEX_NAME}
+      - CONFIG=${ALGOLIA_CONFIG}
+    commands:
+      - pipenv run python -m src.index
```
Comment (tboerger) [inline .drone.yml]: Why does it need this pipenv stuff? Maybe we got to build our own Docker image for that...

**Comment by settermjd** [comment]
Yeah, I'm still working on that.

File: .drone.yml [hunk 8]
```diff
@@ -84,6 +84,25 @@ pipeline:
       local: false
       event: [ push ]
 
+  update-search-index:
+    image: algolia/docsearch-scraper
+    entrypoint: ""
+    pull: true
+    secrets: [
```
Comment (tboerger) [inline .drone.yml]: Please use one or another approach for the formatting of array values, only one of these:

```
secrets: [ a, b, c ]
```

```
secrets:
- a
- b
- c
```

File: .drone.yml (hunk 5 above)
Comment (settermjd) [reply .drone.yml]: Okay, I'll do a bit of reading to work out how to do that.

File: .drone.yml (hunk 7 above)
Comment (settermjd) [reply .drone.yml]: It seemed to be what was required, as I understood it at the time.

File: .drone.yml (hunk 8 above)
Comment (settermjd) [reply .drone.yml]: Just trying to make it look neat and tidy.

File: .drone.yml (hunk 1 above)
Comment (settermjd) [reply .drone.yml]: I dropped the two that seemed okay to drop.

Comment (tboerger) [reply .drone.yml]: The others like `ALGOLIA_API_KEY`, `APPLICATION_ID` and `API_KEY` should also be dropped from environment, just define them as secrets, if it's a secret at all.

File: .drone.yml (hunk 7 above)
Comment (settermjd) [reply .drone.yml]: From re-reading https://community.algolia.com/docsearch/run-your-own.html#running-your-config, the pipeline only needs the docsearch tool available, a Docker image isn't required, along with the config file.

File: .drone.yml (hunk 1 above)
Comment (settermjd) [reply .drone.yml]: They're not really secrets. Thanks for the suggestion.

File: .drone.yml (hunk 7 above)
Comment (settermjd) [reply .drone.yml]: Just made a further update to .drone.yml. How close is it now?

**Pull request merged**